### DISCUSSION TITLE
Dynamic deep links

### DISCUSSION THREAD

**Discussion by PapaGeil** [microsoft-search/pnp-modern-search#3743] [Q&A]
Hi all,

I think the deep links feature is very useful. I am struggling with one thing, though: 

I want to dynamically create URLs with certain filters in my Search Results web part (I know, the documentation tells me not to do this, but the syntax of the deep links is quite straight forward). I am wondering whether there's a way of encoding a given filter string to Base64, so that it's compatible with the URL syntax? I checked on the documentation, but couldn't find anything. The company I'm working for doesn't allow me to write my own javascripts, so I'm hoping for a different solution of some kind.

Thanks!

**Comment by wobba** [comment]
The filter value passed down in the refinement property of a query is hex encoded with two special characters in front `ǂǂ`.
